comb_spec_searcher/strategies/rule.py

```diff
@@ -419,19 +430,19 @@ def eqv_path_rules(self) -> List[Tuple[CombinatorialClassType, Rule]]:
         return eqv_path_rules
 
     def backward_map(
-        self, objs: Tuple[CombinatorialObjectType, ...]
+        self, objs: Tuple[Optional[CombinatorialObjectType], ...]
     ) -> CombinatorialObjectType:
-        res = objs
+        res = cast(Tuple[CombinatorialObjectType], objs)
```

Comment:
Maybe we should enforce the assumption we're doing with the cast with
```python
assert all(obj is not None for obj in objs, "..."
```
What do you think?

Reply:
I don't think a runtime check is necessary.

Reply:
Then I think it's ready to merge